### PR TITLE
Appsflyer : Update regarding support for all sources.

### PR DESCRIPTION
**Pull request by abandonned**
## Description of the change

Providing information about support for all sources. Additionally providing information about transformation that we have to use in dashboard to manipulate the os name.

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Review by abandonned — COMMENTED** [review]
LGTM

File: destinations/attribution/appsflyer.mdx [hunk 1]
```diff
@@ -60,9 +60,14 @@ For more information on where to find the AppsFlyer dev key, refer to the <a hre
 For more information on this field can be found in the <a href="https://support.appsflyer.com/hc/en-us/articles/207034486-Server-to-server-events-API-for-mobile-S2S-mobile-#sharing_filter-16">AppsFlyer documentation</a>.
 </div>
 
-- **App ID**: If you wish to send events to AppsFlyer via the **Android**, **iOS**, or **React Native** source, you must enter the respective **App ID** for each SDK:
-  - **Android App ID** is the application id used in your `app/build.gradle` file
-  - **Apple App ID** is the iTunes Application ID, and it is mandatory for iOS applications.
+- **App ID**: RudderStack supports all the sources for the AppsFlyer destination. If your source is other than **Android** or the **Apple** sources (iOS, watchOS, ipados, tvOS) then transformation in dashboard is needed to manipulate the os name.
```
Comment (saikumarrs) [inline destinations/attribution/appsflyer.mdx:63]:
```suggestion
- **App ID**: RudderStack supports all the sources for the AppsFlyer destination. If your source is other than **Android** or the **Apple** sources (iOS, watchOS, iPadOS, tvOS) then a user transformation in the dashboard is needed to manipulate the `os` name.
```

File: destinations/attribution/appsflyer.mdx [hunk 2]
```diff
@@ -238,6 +243,35 @@ Make sure to use the <code class="inline-code">await</code> keyword with the <co
...
+
+- Go to your [**RudderStack dashboard**](https://app.rudderstack.com/).
+
+- Add a new transformation by following the instructions listed in [**this guide**](https://rudderstack.com/docs/transformation/#adding-a-transformation). Use the following transformation function:
+
+```javascript
+export function transformEvent(event, metadata) {
+  event.context.os = {
+    name: "android"
+  };
```
Comment (saikumarrs) [inline destinations/attribution/appsflyer.mdx:258]: @desusai7 is context.os.version required for AppsFlyer destination? I see that we populate it in the payload

File: destinations/attribution/appsflyer.mdx [hunk 3]
```diff
@@ -60,9 +60,14 @@ For more information on where to find the AppsFlyer dev key, refer to the <a hre
...
-- **App ID**: If you wish to send events to AppsFlyer via the **Android**, **iOS**, or **React Native** source, you must enter the respective **App ID** for each SDK:
-  - **Android App ID** is the application id used in your `app/build.gradle` file
-  - **Apple App ID** is the iTunes Application ID, and it is mandatory for iOS applications.
+- **App ID**: RudderStack supports all the sources for the AppsFlyer destination. If your source is other than **Android** or the **Apple** sources (iOS, watchOS, ipados, tvOS) then transformation in dashboard is needed to manipulate the os name.
+  - **Android App ID**: This the application ID used in your `app/build.gradle` file.
+  - **Apple App ID**: This is the iTunes Application ID and it is mandatory for the iOS applications.
+
+<div class="infoBlock">
+
+RudderStack supports all the other sources for the AppsFlyer destination, in addition to the Android and iOS sources. Refer to the <a href="#connecting-appsflyer-to-non-androidios-sources">Connecting AppsFlyer to non-Android/iOS sources</a> section below for more information.
```
Comment (saikumarrs) [inline destinations/attribution/appsflyer.mdx:69]:
```suggestion
RudderStack supports all the other sources for the AppsFlyer destination, in addition to the Android and iOS sources. Refer to the <a href="#connecting-appsflyer-to-non-androidios-sources">Connecting AppsFlyer to non-Android/Apple sources</a> section below for more information.
```

File: destinations/attribution/appsflyer.mdx [hunk 4]
```diff
@@ -238,6 +243,35 @@ Make sure to use the <code class="inline-code">await</code> keyword with the <co
     </TabPanels>
 </Tabs>
 
+## Connecting AppsFlyer to non-Android/iOS sources
```
Comment (saikumarrs) [inline destinations/attribution/appsflyer.mdx:246]:
```suggestion
## Connecting AppsFlyer to non-Android/Apple sources
```

File: destinations/attribution/appsflyer.mdx (hunk 2 above)
Comment (desusai7) [reply destinations/attribution/appsflyer.mdx:258]: Yeah, we need it for correct data processing as mentioned [here](https://support.appsflyer.com/hc/en-us/articles/207034486-Server-to-server-events-API-for-mobile-S2S-mobile-#os-20)
More details [here](https://support.appsflyer.com/hc/en-us/articles/207034486-Server-to-server-events-API-for-mobile-S2S-mobile-#-required-by-many-ad-networks-for-optimization-purposes-to-enable-correct-data-processing-you-must-send-this-parameter-for-backward-compatibility-reasons-we-dont-enforce-this-so-it-isnt-marked-as-obligatory-22)

Comment (saikumarrs) [reply destinations/attribution/appsflyer.mdx:258]: Thanks, @desusai7. @ameypv-rudder let's update to mention that version is also required.

Comment (abandonned) [reply destinations/attribution/appsflyer.mdx]: have updated doc to include version as well.

File: destinations/attribution/appsflyer.mdx [hunk 5]
```diff
@@ -60,9 +60,14 @@ For more information on where to find the AppsFlyer dev key, refer to the <a hre
 For more information on this field can be found in the <a href="https://support.appsflyer.com/hc/en-us/articles/207034486-Server-to-server-events-API-for-mobile-S2S-mobile-#sharing_filter-16">AppsFlyer documentation</a>.
 </div>
 
-- **App ID**: If you wish to send events to AppsFlyer via the **Android**, **iOS**, or **React Native** source, you must enter the respective **App ID** for each SDK:
-  - **Android App ID** is the application id used in your `app/build.gradle` file
-  - **Apple App ID** is the iTunes Application ID, and it is mandatory for iOS applications.
+- **App ID**: RudderStack supports all the sources for the AppsFlyer destination. If your source is other than **Android** or the **Apple** sources (iOS, watchOS, iPadOS, tvOS) then a user transformation in the dashboard is needed to manipulate the `os` name and version.
```
Comment (saikumarrs) [inline destinations/attribution/appsflyer.mdx:63]:
```suggestion
- **App ID**: RudderStack supports all the sources for the AppsFlyer destination. If your source is other than **Android** or the **Apple** sources (iOS, watchOS, iPadOS, tvOS) then a user transformation in the dashboard is needed to manipulate the OS information.
```

**Review by saikumarrs — APPROVED** [review]

**Review by ameypv-rudder — APPROVED** [review]
LGTM

File: destinations/attribution/appsflyer.mdx [hunk 6]
```diff
@@ -238,6 +245,36 @@ Make sure to use the <code class="inline-code">await</code> keyword with the <co
...
+
+- Go to your [**RudderStack dashboard**](https://app.rudderstack.com/).
+
+- Add a new transformation by following the instructions listed in [**this guide**](https://rudderstack.com/docs/transformation/#adding-a-transformation). Use the following transformation function:
+
+```javascript
+export function transformEvent(event, metadata) {
+  event.context.os = {
+    name: "android",
+    versions: "8.1.0"
```
Comment (saikumarrs) [inline destinations/attribution/appsflyer.mdx:260]:
```suggestion
    version: "8.1.0"
```

**Pull request merged**